src/main/java/org/commcare/formplayer/services/CaseSearchHelper.java

```diff
@@ -32,18 +45,20 @@
 @Component
 public class CaseSearchHelper {
 
+    private static final String CASE_SEARCH_INDEX_TABLE_PREFIX = "case_search_index_storage_";
     @Autowired
     CacheManager cacheManager;
 
     @Autowired
     private RestoreFactory restoreFactory;
-
     @Autowired
     private WebClient webClient;
 
+    @Autowired
+    private FormplayerStorageFactory storageFactory;
     private final Log log = LogFactory.getLog(CaseSearchHelper.class);
 
-    public AbstractTreeElement getExternalRoot(String instanceId, ExternalDataInstanceSource source,
+    public synchronized AbstractTreeElement getExternalRoot(String instanceId, ExternalDataInstanceSource source,
```

Comment:
Why is it necessary to change this to a synchronized method?

Reply:
If a user makes duplicate requests from different tabs, I wanted to make sure that we don't refetch and parse case search results when they are getting already processed into sandbox by another thread. Does this look a right way to do that ?

Reply:
Ahh makes sense. I will revert this. Thanks!